src/index.md

```diff
@@ -985,35 +985,109 @@ For the purposes of brevity this document will use the term "bug" to refer to bo
+
+    $ git checkout master
+    $ git checkout -b my-fix
+
+Here we `checkout` the `master` branch, this is safe to do even if you already have the `master` branch active. Then we create a new branch called `my-fix` and set the repository up to work in that branch.
+
+::: {.box}
+If you're new to git you should read more about how to work with git in one of the many fine git tutorials available on the Internet. This guide doesn't aspire to become another git guide.
+:::
+
+## Generating an SSH key
+
+All pushes require an SSH key which must be installed on GitHub. If this is the first time you clone the [openjdk/jdk](https://github.com/openjdk/jdk) repository you may want to create an SSH key to use with it. For security reasons you should always create new keys and use different keys with each repository you clone. The `ssh-keygen` command generates an SSH key. The `-t` option determines which type of key to create. `ed25519` is recommended. `-C` is used to add a comment in the key file, to help you remember which key it is. While it’s possible to use SSH without a passphrase, this is **strongly discouraged**. Empty or insecure passphrases may be reset using `ssh-keygen -p`; this doesn’t change the keys.
+
+    $ ssh-keygen -t ed25519 -C openjdk-jdk -f ~/.ssh/openjdk-jdk
+    Generating public/private ed25519 key pair.
+    Enter passphrase (empty for no passphrase):
+    Enter same passphrase again:
+    Your identification has been saved in /Users/jesper/.ssh/openjdk-jdk.
+    Your public key has been saved in /Users/jesper/.ssh/openjdk-jdk.pub.
+    The key fingerprint is:
+    SHA256:WS4jCQMtat75ZEue+so+Lgj7V/sdMtj1FTNkfNsCfHA openjdk-jdk
+    The key's randomart image is:
+    +--[ED25519 256]--+
+    |  ..       ..oE  |
+    |  ...       o+o .|
+    | . .o     .  o+.o|
+    |..   o . +    .=.|
+    |o . . o S o   .. |
+    |.. o +.+ + . .   |
+    |o.  *.+.+ . .    |
+    |o....=.  + .     |
+    | .=B=. .. .      |
+    +----[SHA256]-----+
+
+`~/.ssh/openjdk-jdk` is a text file containing your private ssh key. There's a corresponding public key in `~/.ssh/openjdk-jdk.pub` (as detailed in the example above). You should **never** share your private key. The *public* key on the other hand should be uploaded to GitHub. Follow the steps below to do that.
+
+* Go to the GitHub settings for your account by choosing "Settings" in the menu by your avatar in the upper right corner
+* Go to "SSH and GPG keys"
+* Click "New SSH key"
+* Title "OpenJDK" (or something else appropriate)
+* Paste the content of `~/.ssh/openjdk-jdk.pub` into the text field
+  * To get the content of the file you can for instance use `cat ~/.ssh/openjdk-jdk.pub`
+  * It will look something like this: `ssh-ed25519 AAAAC3NzaC1lZDI1NTE5AAAAIO8+egiIgWV+tE7LVVJmlR7WS2Lr3Fj7dXVo9HiasD6T openjdk-jdk`
+* Click "Add SSH key"
+
+Now you are ready to clone your [openjdk/jdk](https://github.com/openjdk/jdk) fork using SSH.
```

Comment:
I suspect this SSH stuff is all fine for Linux and Mac. What about Windows? I imagine that one would need to install OpenSSH or something before this will work. I haven't done this though.

Reply:
I've never done this on Windows. I'll try to get hold of a windows machine to try it out and add a windows section later.